src/config.cpp

```diff
@@ -167,16 +165,15 @@ static void reset() {
 	core_opts = core();
 
 	pattern_opts.clear();
-	source.reset(new rapidjson::Document());
 }
 
-__attribute__((constructor(300)))
```

Comment:
There was no need to change the order of `init_opts()` and `initialize()` functions.